File: Lab1/Temperature/Celsius.cs
namespace Temperature;

class Celsius : Temperature
{
    public Celsius(double degrees) : base(degrees)
    {
        if (degrees < -273.15) throw new ArgumentException("Absolute zero on Celsius is −273.15°C.");
    }

    public Fahrenheit ToFahrenheit()
    {
        return new Fahrenheit(this.value * 1.8 + 32);
    }

    public Kelvin ToKelvin()
    {
        return new Kelvin(this.value + 273.15);
    }
}

File: Lab1/src/Temperature/Celsius.java
package SoftwareEngineering.Lab1.Temperature;

public class Celsius extends BaseTemperature {
    public Celsius(double temperature) {
        // TODO Absolute zero on Celsius is −273.15°C
        super(temperature);
    }

    public Celsius toCelsius() {
        return new Celsius(temperature);
    }

    public Fahrenheit toFahrenheit() {
        return new Fahrenheit(temperature * 1.8 + 32);
    }

    public Kelvin toKelvin() {
        return new Kelvin(temperature + 273.15);
    }
}